api/http/management.go

```diff
@@ -408,67 +409,84 @@ func (h ManagementController) websocketWriter(
 	lastKeystrokeAt := time.Now().UTC().UnixNano()
 Loop:
 	for {
+		var forwardedMsg []byte
+
 		select {
 		case msg := <-deviceChan:
 			mr := &ws.ProtoMsg{}
 			err = msgpack.Unmarshal(msg.Data, mr)
 			if err != nil {
 				return err
 			}
+
+			forwardedMsg = msg.Data
+
 			if mr.Header.Proto == ws.ProtoTypeShell {
 				switch mr.Header.MsgType {
 				case shell.MessageTypeShellCommand:
-					timeNowUTC := time.Now().UTC().UnixNano()
-					keystrokeDelay := timeNowUTC - lastKeystrokeAt
-					if keystrokeDelay >= keyStrokeDelayRecordingThresholdNs {
-						if keystrokeDelay > keyStrokeMaxDelayRecording {
-							keystrokeDelay = keyStrokeMaxDelayRecording
+
+					if recordedBytes >= app.MessageSizeLimit || controlBytes >= app.MessageSizeLimit {
+						sendLimitErrDevice(ctx, session, h.nats)
```

Comment:
connected to the below comment, same thing, I kind of would not send the message on every byte over the session bytes transferred limit. what if we close the session and ignore all messages above the limit? 
also have you looked at the 
```go
func (a *app) FreeUserSession(
	ctx context.Context,
	sessionID string,
) error
```
are you sure that the session is freed when it reaches the limit?

Reply:
hmm, I though I am freeing everything - because when we receive a stop from the device, the sessions are cleaned up in the device portion of the api...
I see that I missed this one func